soccer-records-model/src/main/java/cz/muni/fi/pa165/team/TeamMatchService.java

```diff
@@ -0,0 +1,195 @@
+     *                             set up as home team in match which should be created
+     * @param allMatchesOfAwayTeam matches in which appears team which should be
+     *                             set up as away team in match which should be created
+     * @param homeTeam home team of match which should be created
+     * @param awayTeam away team of match which should be created
+     * @param startTime start time of match which should be created
+     * @param endTime end time of match which should be created
+     * @return newly created match
+     */
+    public TeamMatch createMatch(
+        List<TeamMatch> allMatchesOfHomeTeam,
+        List<TeamMatch> allMatchesOfAwayTeam,
+        Team homeTeam,
+        Team awayTeam,
+        Date startTime,
+        Date endTime
+    )
+    {
+        Assert.notNull(homeTeam, "Match cannot be created with a null home team");
+        Assert.notNull(awayTeam, "Match cannot be created with a null away team");
+        Assert.isTrue(!homeTeam.equals(awayTeam), "Match cannot be created for home and away teams which are same");
+
+        validateMatchTimes(startTime, endTime);
+
+        if (!allMatchesOfHomeTeam.isEmpty()) {
+            for (TeamMatch tm : allMatchesOfHomeTeam) {
+                if (startTime.equals(tm.getStartTime())) {
+                    throw new MatchWithSameParametersAlreadyExistsException(
+                        homeTeam.getId(),
+                        startTime
+                    );
+                }
+            }
+        }
+
+        if (!allMatchesOfAwayTeam.isEmpty()) {
+            for (TeamMatch tm : allMatchesOfAwayTeam) {
+                if (startTime.equals(tm.getStartTime())) {
+                    throw new MatchWithSameParametersAlreadyExistsException(
+                        awayTeam.getId(),
+                        startTime
+                    );
+                }
+            }
+        }
+
+        return new TeamMatch(homeTeam, awayTeam, startTime, endTime);
+    }
+
+    /**
+     * This method changes start time and end time of match if new times are valid.
+     *
+     * @param sameStartTimeMatches matches which has same start time as new start time
+     *                             which should be updated at the match
+     * @param match match which start time and end time should be updated
+     * @param startTime new start time of the match
+     * @param endTime new end time of the match
+     */
+    public void changeMatchTime(
+        List<TeamMatch> sameStartTimeMatches,
+        TeamMatch match,
+        Date startTime,
+        Date endTime
+    )
+    {
+        Assert.notNull(match, "Cannot change match times to null match");
+        validateMatchTimes(startTime, endTime);
+
+        if (!sameStartTimeMatches.isEmpty()) {
+            for (TeamMatch tm : sameStartTimeMatches) {
+                if (match.getHomeTeam().equals(tm.getHomeTeam()) ||
```

Comment:
tyhle dva "velky" ify taky... je to uplne stejna logika, jenom jinak parametrizovana